keras_cv/models/object_detection/faster_rcnn.py

```diff
@@ -0,0 +1,326 @@
+            return call_single_level(feature_map)
+        elif isinstance(feature_map, (list, tuple)):
+            rpn_boxes = []
+            rpn_scores = []
+            for f_map in feature_map:
+                rpn_box, rpn_score = call_single_level(f_map)
+                rpn_boxes.append(rpn_box)
+                rpn_scores.append(rpn_score)
+            rpn_boxes = tf.concat(rpn_boxes, axis=1)
+            rpn_scores = tf.concat(rpn_scores, axis=1)
+            return rpn_boxes, rpn_scores
+        else:
+            rpn_boxes = {}
+            rpn_scores = {}
+            for lvl, f_map in feature_map.items():
+                rpn_box, rpn_score = call_single_level(f_map)
+                rpn_boxes[lvl] = rpn_box
+                rpn_scores[lvl] = rpn_score
+            rpn_boxes = tf.concat(tf.nest.flatten(rpn_boxes), axis=1)
+            rpn_scores = tf.concat(tf.nest.flatten(rpn_scores), axis=1)
+            return rpn_boxes, rpn_scores
+
+    def get_config(self):
+        config = {
+            "num_anchors_per_location": self.num_anchors,
+        }
+        base_config = super().get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+# class agnostic regression
+class RCNNHead(tf.keras.layers.Layer):
+    def __init__(
+        self,
+        classes,
+        conv_dims=[],
+        fc_dims=[1024, 1024],
+        **kwargs,
+    ):
+        super(RCNNHead, self).__init__(**kwargs)
+        self.num_classes = classes
+        self.conv_dims = conv_dims
+        self.fc_dims = fc_dims
+        self.convs = []
+        for conv_dim in conv_dims:
+            layer = tf.keras.layers.Conv2D(
+                filters=conv_dim,
+                kernel_size=3,
+                strides=1,
+                padding="same",
+                activation="relu",
+            )
+            self.convs.append(layer)
+        self.fcs = []
+        for fc_dim in fc_dims:
+            layer = tf.keras.layers.Dense(units=fc_dim, activation="relu")
+            self.fcs.append(layer)
+        self.box_pred = tf.keras.layers.Dense(units=4)
+        self.cls_score = tf.keras.layers.Dense(units=classes + 1)
+
+    def call(self, feature_map):
+        x = feature_map
+        for conv in self.convs:
+            x = conv(x)
+        for fc in self.fcs:
+            x = fc(x)
+        rcnn_boxes = self.box_pred(x)
+        rcnn_scores = self.cls_score(x)
+        return rcnn_boxes, rcnn_scores
+
+    def get_config(self):
+        config = {
+            "classes": self.num_classes,
+            "conv_dims": self.conv_dims,
+            "fc_dims": self.fc_dims,
+        }
+        base_config = super().get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+# TODO(tanzheny): add more configurations
+class FasterRCNN(tf.keras.Model):
+    def __init__(
+        self,
+        classes,
+        bounding_box_format,
+        backbone=None,
```

Comment:
lets make sure to include support_rescaling here too as a required argument

Reply:
Done.